openhands/server/routes/settings.py

```diff
@@ -42,19 +40,11 @@ async def load_settings(
                 content={'error': 'Settings not found'},
             )
 
-        provider_tokens_set = {}
-
-        if bool(user_id):
-            provider_tokens_set[ProviderType.GITHUB.value] = True
-
+        provider_tokens_set: dict[ProviderType, str | None]  = {}
         if provider_tokens:
-            all_provider_types = [provider.value for provider in ProviderType]
-            provider_tokens_types = [provider.value for provider in provider_tokens]
-            for provider_type in all_provider_types:
-                if provider_type in provider_tokens_types:
-                    provider_tokens_set[provider_type] = True
-                else:
-                    provider_tokens_set[provider_type] = False
+            for provider_type, provider_token in provider_tokens.items():
+                if provider_token.token or provider_token.user_id:
```

Comment:
This is the BE logic where we only add provider type information if it is currently set